nilearn/decoding/decoder.py

```diff
@@ -410,8 +410,12 @@ def fit(self, X, y, groups=None):
 
         # Setup scorer
         scorer = check_scoring(self.estimator, self.scoring)
-        self.cv_ = list(check_cv(self.cv, y=y,
-            classifier=self.is_classification).split(X, y, groups=groups))
+
+        # Setup cross-validation object
+        if cv == int or cv is None and groups is not None:
+            self.cv_ = list(
+                check_cv(self.cv, y=y,
+                         classifier=self.is_classification).split(X, y, groups=groups))
```

Comment:
Assign this to a variable with a sensible name and then assign that variable as `self.cv_ = list(var_sensible_name)`

Reply:
My apologies but the fix for cv group is not finished yet - I accidentally added the `decoder.py` file when trying to have the decoder2 branch up to date with Nilearn upstream. Will finish this with a test case very soon.

Reply:
You mean you want to remove the `decoder.py` file from this PR?